stac_fastapi/types/setup.py

```diff
@@ -6,7 +6,7 @@
     desc = f.read()
 
 install_requires = [
-    "fastapi>=0.73.0,<0.87",
+    "fastapi>=0.73.0",
```

Comment:
FYI 0.95 breaks quite some stuff in TiTiler (also 0.94 broke stuff with the new starlette version)

Reply:
Can you point me at an example or two? Doesn't look like the test suite for **stac-fastapi** catches those problems as it currently stands ... it'd be nice to push the tests into here so we can fix as needed.

Unless it's just a **TiTiler** thing, and we should let implementations ceil **fastapi** where they need?

Reply:
👍 It might be only a TiTiler thing for sure. Issues were:
- 0.94 updated starlette and now the `request.url_for` methods returns a URL object (it was a string before): https://github.com/developmentseed/titiler/pull/607
- 0.95 added the `Annotated` type support and refactored most of the `fastapi.params` which now breaks titiler because we used Optional Path parameter (https://github.com/tiangolo/fastapi/discussions/9278)

There has been also other issues reported in https://github.com/tiangolo/fastapi/discussions since the release.

Again, just an FYI so you don't spend too long if the tests were failing ... but they seem to pass 😄

Reply:
Awesome, thanks. I'm inclined to 🙈 on those problems for this repo, and leave it to implementers to bound their **fastapi** versions as they see fit (folks _should_ be pinning to a specific version of all the deps for deployed instances, anyways, right?!).

Reply:
> folks should be pinning to a specific version of all the deps for deployed instances, anyways, right?

🙈